### PR TITLE
Build multi-platform snapshot images

### LINKED ISSUE #5768: Build multi-platform snapshot images
### Problem

Currently snapshot images are being created and deployed in the registry, however they are build only for `amd64` platform. This causes inconvenience for people working on `arm64`, because those images are not working correctly, so they have to be build manually.

### Solution

Enhance current release setup, to support cross-build.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds multi-platform snapshot image build to `release-integration` github workflow and optimizes `rest`, `rest-monitor`, and `rosetta` image build time

- Build multi-platform snapshot images
- Enable rosetta cross-platform build
- Switch to `@trufflesuite/bigint-buffer` with prebuilt multi-platform binaries
- Optimize `rest` and `rest-monitor` Dockerfile

**Related issue(s)**:

Fixes #5768 

**Notes for reviewer**:

There will be warning when we run the new rest image built for linux-arm64 on m1 macs:

`bigint: Failed to load bindings, pure JS will be used (try npm run rebuild?)`

That's because the new package only provides arm64 prebuilt binary for `macosx` while the image built expects linux-arm64 binary. Further more, the project `@trufflesuite/bigint-buffer` doesn't provide a linux-arm64 binary because github doesn't provide a `linux-arm64` runner. 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/release-integration.yml [hunk 1]
```diff
@@ -6,10 +6,23 @@ on:
...
+        module: [ graphql, grpc, importer, monitor, rest, rest-monitor, rosetta, web3 ]
+    env:
+      CONTEXT: hedera-mirror-${{ matrix.module }}
+      IMAGE: gcr.io/mirrornode/hedera-mirror-${{ matrix.module }}
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
 
+      - name: Get version
+        run: echo "VERSION=$(grep -oP 'version=\K.+' gradle.properties)" >> $GITHUB_ENV
```
Comment (xin-hedera) [inline .github/workflows/release-integration.yml:20]: reads the snapshot version from `gradle.properties`

File: hedera-mirror-rosetta/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,12 @@
-FROM golang:1.20.2-bullseye as build
+FROM --platform=$BUILDPLATFORM golang:1.20.2-bullseye as build
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/Dockerfile:1]: Shouldn't this be TARGETPLATFORM?
```suggestion
FROM --platform=$TARGETPLATFORM golang:1.20.2-bullseye as build
```

Comment (xin-hedera) [reply hedera-mirror-rosetta/Dockerfile:1]: it should be `BUILDPLATFORM`, so we can cross-compile arm64 binaries on amd64 platform without emulating an arm64 build container on amd64 build host

**Comment by xin-hedera** [comment]
chart install workflow job failure due to env variables are only available with docker `buildkit` plugin, will try to fix

File: hedera-mirror-rosetta/Dockerfile (hunk 2 above)
Comment (jascks) [reply hedera-mirror-rosetta/Dockerfile:1]: I'm not sure if this is required for various reasons given Go, GCP, building the image using gradle on Github etc., but when I recently created the new PG15+Citus images using `docker buildx` to build a multi-platform Alpine images, I did not modify the `Dockerfile` itself with platform specific changes.

After setting up the build to use with buildx, I simply specified `--platform linux/arm64,linux/amd64` and pushed and that resulted in a single tag on Docker Hub, with OS/ARCH indicated for both platforms.

Comment (xin-hedera) [reply hedera-mirror-rosetta/Dockerfile:1]: it's performance optimization, reduces the multi-platform image build time for rosetta in our github workflow from 15m to 7m

Comment (jascks) [reply hedera-mirror-rosetta/Dockerfile:1]: Ok, nice!

**Review by jascks — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**